File: app/src/main/java/org/pyload/android/client/dialogs/FileInfoDialog.java
package org.pyload.android.client.dialogs;

import android.app.Dialog;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import org.pyload.android.client.R;
import org.pyload.thrift.FileData;
import org.pyload.thrift.PackageData;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

public class FileInfoDialog extends DialogFragment {

    private PackageData pack = null;
    private FileData file = null;

    public static FileInfoDialog newInstance(PackageData pack, FileData file) {

        FileInfoDialog dialog = new FileInfoDialog();
        Bundle args = new Bundle();

        args.putSerializable("pack", pack);
        args.putSerializable("file", file);

        dialog.setArguments(args);
        return dialog;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        pack = (PackageData) getArguments().getSerializable("pack");
        file = (FileData) getArguments().getSerializable("file");
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View dialog = inflater.inflate(R.layout.fileinfo_dialog, container, false);

        TextView view = dialog.findViewById(R.id.name);
        view.setText(file.name);

        view = dialog.findViewById(R.id.status);
        view.setText(file.statusmsg);

        view = dialog.findViewById(R.id.plugin);
        view.setText(file.plugin);

        view = dialog.findViewById(R.id.size);
        view.setText(file.format_size);

        view = dialog.findViewById(R.id.error);
        view.setText(file.error);

        view = dialog.findViewById(R.id.packageValue);
        view.setText(pack.name);

        view = dialog.findViewById(R.id.folder);
        view.setText(pack.folder);

        Button button = dialog.findViewById(R.id.close);
        button.setOnClickListener(v -> dismiss());

        return dialog;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Dialog dialog = super.onCreateDialog(savedInstanceState);
        dialog.setTitle(R.string.fileinfo_title);
        return dialog;
    }
}

File: app/src/main/java/org/pyload/android/client/dialogs/FileInfoDialog.kt
package org.pyload.android.client.dialogs

import android.app.Dialog
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import org.pyload.android.client.R
import org.pyload.thrift.FileData
import org.pyload.thrift.PackageData

class FileInfoDialog : DialogFragment() {
    private lateinit var pack: PackageData
    private lateinit var file: FileData

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        pack = arguments?.getSerializable("pack") as PackageData
        file = arguments?.getSerializable("file") as FileData
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val dialog = inflater.inflate(R.layout.fileinfo_dialog, container, false)

        dialog.findViewById<TextView>(R.id.name).text = file.name
        dialog.findViewById<TextView>(R.id.status).text = file.statusmsg
        dialog.findViewById<TextView>(R.id.plugin).text = file.plugin
        dialog.findViewById<TextView>(R.id.size).text = file.format_size
        dialog.findViewById<TextView>(R.id.error).text = file.error
        dialog.findViewById<TextView>(R.id.packageValue).text = pack.name
        dialog.findViewById<TextView>(R.id.folder).text = pack.folder
        dialog.findViewById<Button>(R.id.close).setOnClickListener { dismiss() }

        return dialog
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val dialog = super.onCreateDialog(savedInstanceState)
        dialog.setTitle(R.string.fileinfo_title)
        return dialog
    }

    companion object {

        fun newInstance(pack: PackageData, file: FileData): FileInfoDialog {
            val dialog = FileInfoDialog()
            val args = Bundle()

            args.putSerializable("pack", pack)
            args.putSerializable("file", file)

            dialog.arguments = args
            return dialog
        }
    }
}